src/context/AppProvider/hooks/useAccountsFetchQuery.tsx

```diff
@@ -0,0 +1,209 @@
+
+          const { hasActivity } = account.accounts.byId[accountId]
+
+          const accountNumberHasChainActivity = !isUtxoChainId(chainId)
+            ? hasActivity
+            : // For UTXO AccountIds, we need to check if *any* of the scriptTypes have activity, not only the current one
+              // else, we might end up with partial account data, with only the first 1 or 2 out of 3 scriptTypes
+              // being upserted for BTC and LTC
+              results.some((res, _idx) => {
+                if (res.status === 'rejected') return false
+                const { data: account } = res.value
+                if (!account) return false
+                const accountId = accountIds[_idx]
+                const { chainId: _chainId } = fromAccountId(accountId)
+                if (chainId !== _chainId) return false
+                return account.accounts.byId[accountId].hasActivity
+              })
+
+          // don't add accounts with no activity past account 0
+          if (accountNumber > 0 && !accountNumberHasChainActivity) {
+            chainIdsWithActivity.delete(chainId)
+            delete accountMetadataByAccountId[accountId]
+          } else {
+            // handle utxo chains with multiple account types per account
+            chainIdsWithActivity.add(chainId)
+
+            dispatch(portfolio.actions.upsertPortfolio(account))
+            const chainIdAccountMetadata = Object.entries(accountMetadataByAccountId).reduce(
+              (acc, [accountId, metadata]) => {
+                const { chainId: _chainId } = fromAccountId(accountId)
+                if (chainId === _chainId) {
+                  acc[accountId] = metadata
+                }
+                return acc
+              },
+              {} as AccountMetadataById,
+            )
+            for (const accountId of Object.keys(chainIdAccountMetadata)) {
+              dispatch(portfolio.actions.enableAccountId(accountId))
+            }
+            dispatch(
+              portfolio.actions.upsertAccountMetadata({
+                accountMetadataByAccountId: chainIdAccountMetadata,
+                walletId,
+              }),
+            )
+          }
+        })
+
+        return results
+      })
+
+      await Promise.allSettled(accountNumberAccountIdsPromises)
+      chainIds = chainIdsWithActivity
+    }
+
+    // Only fetch and upsert Tx history once all are loaded, otherwise big main thread rug
+    const { getAllTxHistory } = txHistoryApi.endpoints
+
+    await Promise.all(
+      enabledWalletAccountIds.map(requestedAccountId =>
+        dispatch(getAllTxHistory.initiate(requestedAccountId)),
+      ),
+    )
+  }, [dispatch, enabledWalletAccountIds, isSnapInstalled, supportedChains, wallet])
+
+  const query = useQuery({
+    queryKey: [
+      'useAccountsFetch',
+      {
+        deviceId,
+        supportedChains,
+      },
+    ],
+    queryFn: deviceId && !hasManagedAccounts ? queryFn : skipToken,
+  })
```

Comment:
Main takeaways since diff won't be obvious is logic is the same as previously, with two notable differences:

- We only enable the query if there is a `deviceId` and no `managedAccounts`, avoiding costly re-runs of async flow (see inline comment above on why this would re-run because of account management flow, and why it still will, though now as a side-effect)
- We've extracted the managed accounts flow as a side-effect
- We keep dependencies to the bare minimum needed `queryKey`-wise to avoid re-runs of async flow, though still leveraging hook closure for fresh refs